Jint/Options.cs

```diff
@@ -80,6 +87,45 @@ public Options AddObjectConverter(IObjectConverter objectConverter)
             return this;
         }
 
+        public Options AddExtensionMethods(params Type[] types)
+        {
+            _extensionMethodClassTypes.AddRange(types);
+            _extensionMethods = ExtensionMethodCache.Build(_extensionMethodClassTypes);
+            return this;
+        }
+
+        private void AttachExtensionMethodsToPrototypes(Engine engine)
+        {
+            AttachExtensionMethodsToPrototype(engine, engine.Array.PrototypeObject, typeof(Array));
+            AttachExtensionMethodsToPrototype(engine, engine.Boolean.PrototypeObject, typeof(bool));
+            AttachExtensionMethodsToPrototype(engine, engine.Date.PrototypeObject, typeof(DateTime));
+            AttachExtensionMethodsToPrototype(engine, engine.Number.PrototypeObject, typeof(double));
+            AttachExtensionMethodsToPrototype(engine, engine.Object.PrototypeObject, typeof(ExpandoObject));
+            AttachExtensionMethodsToPrototype(engine, engine.RegExp.PrototypeObject, typeof(System.Text.RegularExpressions.Regex));
+            AttachExtensionMethodsToPrototype(engine, engine.String.PrototypeObject, typeof(string));
+        }
+
+        private void AttachExtensionMethodsToPrototype(Engine engine, ObjectInstance prototype, Type objectType)
+        {
+            if (!_extensionMethods.TryGetExtensionMethods(objectType, out var methods))
+            {
+                return;
+            }
+
+            foreach (var overloads in methods.GroupBy(x => x.Name))
+            {
+                var functionInstance = new MethodInfoFunctionInstance(engine, MethodDescriptor.Build(overloads.ToList()));
+                var descriptor = new PropertyDescriptor(functionInstance, PropertyFlag.None);
+
+                // make sure we register both lower case and upper case
+                prototype.SetOwnProperty(overloads.Key, descriptor);
+                if (char.IsUpper(overloads.Key[0]))
```

Comment:
shouldn't it be a method resolution logic? Do we do that for standard methods?

Reply:
Unfortunately that only applies to `ObjectWrapper` which does first-char-case-insensitive logic at the moment. We cannot bend the JS spec the same way. Maybe later this could be configurable if someone needs such option.

Reply:
Ok